src/Bridges/Constraint/bridges/semi_to_binary.jl

```diff
@@ -75,6 +75,19 @@ function bridge_constraint(
     f::MOI.VariableIndex,
     s::S,
 ) where {T<:Real,S<:Union{MOI.Semicontinuous{T},MOI.Semiinteger{T}}}
+    F = MOI.VariableIndex
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.GreaterThan{T}}(f.value))
+        throw(MOI.LowerBoundAlreadySet{MOI.GreaterThan{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.LessThan{T}}(f.value))
+        throw(MOI.UpperBoundAlreadySet{MOI.LessThan{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.EqualTo{T}}(f.value))
+        throw(MOI.LowerBoundAlreadySet{MOI.EqualTo{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.Interval{T}}(f.value))
+        throw(MOI.LowerBoundAlreadySet{MOI.Interval{T},S}(f))
+    end
```

Comment:
This part is more controversial: https://github.com/jump-dev/MathOptInterface.jl/pull/1901#discussion_r908153753

I added these because if a bound already exists, then we need to error that adding Semi-set isn't allowed. If we want until later, we don't know in which order they were added.

On reflection though, I think this is actually a bug in the `AbstractBridgeOptimizer`, which should catch this at the modeling level instead of the bridge (the bridge can do what it likes, it's just illegal to add at the model level).